src/languageTools/LanguageClientWrapper.js

```diff
@@ -0,0 +1,620 @@
+            return requestHandler.call(null, params.params);
+        }
+        console.log("No handler provided for server request type : ", params.type);
+        return $.Deferred().reject();
+
+    };
+
+    LanguageClientWrapper.prototype._onNotificationDelegator = function (params) {
+        if (!params || !params.type) {
+            console.log("Invalid server notification");
+            return;
+        }
+
+        var notificationHandlers = this._onNotificationHandlers[params.type];
+        if (notificationHandlers && Array.isArray(notificationHandlers) && notificationHandlers.length) {
+            notificationHandlers.forEach(function (handler) {
+                if (validateHandler(handler)) {
+                    handler.call(null, params.params);
+                }
+            });
+        } else {
+            console.log("No handlers provided for server notification type : ", params.type);
+        }
+    };
+
+    LanguageClientWrapper.prototype._request = function (type, params) {
+        params = validateRequestParams(type, params);
+        if (params) {
+            params = _addTypeInformation(type, params);
+            return this._requestClient(params);
+        }
+
+        console.log("Invalid Parameters provided for request type : ", type);
+        return $.Deferred().reject();
+    };
+
+    LanguageClientWrapper.prototype._notify = function (type, params) {
+        params = validateNotificationParams(type, params);
+        if (params) {
+            params = _addTypeInformation(type, params);
+            this._notifyClient(params);
+        } else {
+            console.log("Invalid Parameters provided for notification type : ", type);
+        }
+    };
+
+    LanguageClientWrapper.prototype._addOnRequestHandler = function (type, handler) {
+        if (validateHandler(handler)) {
+            this._onRequestHandler[type] = handler;
+        }
+    };
+
+    LanguageClientWrapper.prototype._addOnNotificationHandler = function (type, handler) {
+        if (validateHandler(handler)) {
+            if (!this._onNotificationHandlers[type]) {
+                this._onNotificationHandlers[type] = [];
+            }
+
+            this._onNotificationHandlers[type].push(handler);
+        }
+    };
+
+    /**
+        Requests
+    */
+    //start
+    LanguageClientWrapper.prototype.start = function (params) {
+        params = validateRequestParams(ToolingInfo.LANGUAGE_SERVICE.START, params);
+        if (params) {
+            return this._startClient(params);
+        }
+
+        console.log("Invalid Parameters provided for request type : start");
+        return $.Deferred().reject();
+    };
+
+    //shutdown
+    LanguageClientWrapper.prototype.stop = function () {
+        return this._stopClient();
+    };
+
+    //restart
+    LanguageClientWrapper.prototype.restart = function (params) {
+        var self = this;
+        return this.stop().done(function () {
```

Comment:
Check this once , restart not resolving with start result

Reply:
fixed